pkg/k8s/secrets_store.go

```diff
@@ -0,0 +1,90 @@
+package k8s
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+)
+
+// NewSecretsStore constructs new conversionStore
+func NewSecretsStore(eventHandler handler.EventHandler, keyFunc cache.KeyFunc, queue workqueue.RateLimitingInterface) *SecretsStore {
+	return &SecretsStore{
+		eventHandler: eventHandler,
+		queue:        queue,
+		store:        cache.NewStore(keyFunc),
+	}
+}
+
+var _ cache.Store = &SecretsStore{}
+
+// SecretsStore implements cache.Store.
+// It invokes the eventhandler for Add, Update, Delete events
+type SecretsStore struct {
+	store        cache.Store
+	queue        workqueue.RateLimitingInterface
+	eventHandler handler.EventHandler
+}
+
+// Add adds the given object to the accumulator associated with the given object's key
+func (s *SecretsStore) Add(obj interface{}) error {
+	if err := s.store.Add(obj); err != nil {
+		return err
+	}
+	s.eventHandler.Create(event.CreateEvent{Object: obj.(*corev1.Secret)}, s.queue)
```

Comment:
i prefer to use a channel, and make the controller watch events from the channel instead.

Reply:
updated